docs/source/reference/types.rst

```diff
@@ -0,0 +1,163 @@
+Types, Functions, and Interfaces
+================================
+
+The type system in Idris 2 is quite powerful, and is the main feature that sets it apart from simply typed languages like Haskell or OCaml.
+
+There are many ways to construct types, those being members of ``Type``, in Idris.
+
+-------------
+Functions
+-------------
+
+Function types, formed by ``(x : A) -> B``, are valid types, given that ``A`` is a valid type *or* kind and ``B`` is a valid type *or* kind. 
```

Comment:
```suggestion
Function types, formed by ``(x : a) -> b``, are valid types, given that ``a``  and ``b`` are valid types.
```

protip: use lower case for generic names for types since they are automatically quantified in the compiler. It should probably be written in this file that `map : (a -> b) -> List a -> List b` is actually `map : {0 a, b : Type} -> (a -> b) -> List a -> List b` because `a` and `b` are lowercase but `List` is uppercase so it's not automatically added as an argument.

Also Idris does not have kinds, only types.